apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx

```diff
@@ -14,7 +14,12 @@ import i18n from '@cdo/locale';
 import {LtiProviderContext} from '../../context';
 
 const ExistingAccountCard = () => {
-  const {ltiProviderName, existingAccountUrl} = useContext(LtiProviderContext)!;
+  const {ltiProvider, ltiProviderName, existingAccountUrl} =
+    useContext(LtiProviderContext)!;
+  const urlParams = new URLSearchParams({
+    lms_name: ltiProviderName,
+    lti_provider: ltiProvider,
+  });
 
```

Comment:
What do you think of this?

```
diff --git a/apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx b/apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx
index d8d8a6b2fea..f0f70b9ca52 100644
--- a/apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx
+++ b/apps/src/lib/ui/lti/link/LtiLinkAccountPage/cards/ExistingAccountCard/index.tsx
@@ -16,10 +16,12 @@ import {LtiProviderContext} from '../../context';
 const ExistingAccountCard = () => {
   const {ltiProvider, ltiProviderName, existingAccountUrl} =
     useContext(LtiProviderContext)!;
+  const callToActionUrl = new URL(existingAccountUrl);
   const urlParams = new URLSearchParams({
     lms_name: ltiProviderName,
     lti_provider: ltiProvider,
   });
+  callToActionUrl.search = urlParams.toString();
 
   return (
     <Card data-testid={'existing-account-card'}>
@@ -43,7 +45,7 @@ const ExistingAccountCard = () => {
           color={buttonColors.purple}
           type={'primary'}
           size="l"
-          href={`${existingAccountUrl}?${urlParams.toString()}`}
+          href={callToActionUrl.href}
           text={i18n.ltiLinkAccountExistingAccountCardActionLabel()}
         />
       </CardActions>
diff --git a/apps/test/unit/lib/ui/lti/link/LtiLinkAccountPage/LtiLinkAccountPageTest.tsx b/apps/test/unit/lib/ui/lti/link/LtiLinkAccountPage/LtiLinkAccountPageTest.tsx
index 10237f182e7..5abcd27da68 100644
--- a/apps/test/unit/lib/ui/lti/link/LtiLinkAccountPage/LtiLinkAccountPageTest.tsx
+++ b/apps/test/unit/lib/ui/lti/link/LtiLinkAccountPage/LtiLinkAccountPageTest.tsx
@@ -12,7 +12,7 @@ const DEFAULT_CONTEXT: LtiProviderContextProps = {
   ltiProvider: 'canvas_cloud',
   ltiProviderName: 'Canvas',
   newAccountUrl: '/new-account',
-  existingAccountUrl: '/existing-account',
+  existingAccountUrl: 'https://test.code.org/existing-account',
 };
 
 describe('LTI Link Account Page Tests', () => {
@@ -41,7 +41,9 @@ describe('LTI Link Account Page Tests', () => {
           .getByText(i18n.ltiLinkAccountExistingAccountCardActionLabel())
           .closest('a')!
           .getAttribute('href')
-      ).to.equal('/existing-account?lms_name=Canvas&lti_provider=canvas_cloud');
+      ).to.equal(
+        'https://test.code.org/existing-account?lms_name=Canvas&lti_provider=canvas_cloud'
+      );
     });
   });
```

Reply:
Nice, much cleaner than cludging the url together in a template string!

Reply:
@stephenliang What do you think about [this](https://github.com/code-dot-org/code-dot-org/pull/58823/commits/acc1a870c5bb2bdb7d2d04de7909d31f0428e6ec)? Makes existingAccountUrl a URL to begin with in the context props. Cleaner?

Reply:
That looks even better!